src/Workspaces/Core/Portable/VirtualChars/AbstractVirtualCharService.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.VirtualChars
+{
+    internal abstract class AbstractVirtualCharService : IVirtualCharService
+    {
+        protected abstract ImmutableArray<VirtualChar> TryConvertToVirtualCharsWorker(SyntaxToken token);
+
+        public ImmutableArray<VirtualChar> TryConvertToVirtualChars(SyntaxToken token)
```

Comment:
@CyrusNajmabadi 
Add a doc comment, so the consumer understands what aspect of the syntax node is being worked on. eg. Is it the actual text from source, or value text, or displayed text?

Reply:
Comment added.

Reply:
#fixed